subprojects/packagefiles/sfml/src/SFML/Graphics/meson.build

```diff
@@ -40,10 +40,12 @@ graphics_sources += files(
     'RenderTextureImplDefault.cpp',
 )
 
+stb = include_directories('../../../extlibs/headers/stb_image')
```

Comment:
Wait, that look for headers outside of the subproject itself? That's not allowed, I'm curious if the CI will tell.

The correct way for making this work is the project that vendor headers should do:
```meson
incdir = include_directories('extlibs/headers/stb_image')
stb_dep = declare_dependency(include_directories: incdir)
meson.override_dependency('stb', stb_dep)
```

That way when sfml subproject lookup for `stb` dependency it will get stb_dep defined by the main project instead of looking on the system.

Reply:
> Wait, that look for headers outside of the subproject itself? That's not allowed, 

No, course not. This goes up of three levels, so to the root of SFML sources. That's allowed.

> I'm curious if the CI will tell.

The CI fails on OpenAL... but it's working fine on my system without STB installed. 

> The correct way...

Mine's correct and simply works. I have no strong opinion in favour or disfavour of one or another, so if overriding the dependency is the recommended way, I'm with that.
I do think that removing the STB dependency declaration from the root meson file helps by constraining the declaration to the module that's using it. 

> That way when sfml subproject lookup for stb dependency it will get stb_dep defined by the main project instead of looking on the system.

There is no "main" project here, `stb_dep` is defined in `sfml/meson.build`, in the meson overlay for SFML itself, not in the root project. But that's just nitpicking on my side.

Reply:
Sorry, my mistake, those headers are indeed inside sfml, I got the number of `..` levels wrong :)

Reply:
Well, you can also just declare `stb_inc = include_directories('extlibs/headers/stb_image')` in the top-level meson.build and reuse that variable in `src/SFML/Graphics/meson.build`, which has the advantage of not adding literal `-I../../../` to the compiler command line AND is clearly less confusing for people reading this PR.